agent/connect/ca/testing.go

```diff
@@ -238,3 +232,129 @@ func requireTrailingNewline(t testing.T, leafPEM string) {
 		t.Fatalf("cert do not end with a new line")
 	}
 }
+
+// The zero value implies unprivileged.
+type VaultTokenAttributes struct {
+	RootPath, IntermediatePath string
+
+	ConsulManaged bool
+	VaultManaged  bool
+	WithSudo      bool
+
+	CustomRules string
+}
+
+func (a *VaultTokenAttributes) DisplayName() string {
+	switch {
+	case a == nil:
+		return "unprivileged"
+	case a.CustomRules != "":
+		return "custom"
+	case a.ConsulManaged:
+		return "consul-managed"
+	case a.VaultManaged:
+		return "vault-managed"
+	default:
+		return "unprivileged"
+	}
+}
+
+func (a *VaultTokenAttributes) Rules(t testing.T) string {
+	switch {
+	case a == nil:
+		return ""
+
+	case a.CustomRules != "":
+		return a.CustomRules
+
+	case a.RootPath == "":
+		t.Fatal("missing required RootPath")
+		return "" // dead code
+
+	case a.IntermediatePath == "":
+		t.Fatal("missing required IntermediatePath")
+		return "" // dead code
+
+	case a.ConsulManaged:
+		// Consul Managed PKI Mounts
+		rules := fmt.Sprintf(`
+path "sys/mounts" {
+  capabilities = [ "read" ]
+}
+
+path "sys/mounts/%[1]s" {
+  capabilities = [ "create", "read", "update", "delete", "list" ]
+}
+
+path "sys/mounts/%[2]s" {
+  capabilities = [ "create", "read", "update", "delete", "list" ]
+}
+
+# Needed for Consul 1.11+
+path "sys/mounts/%[2]s/tune" {
+  capabilities = [ "update" ]
+}
+
+# vault token renewal
+path "auth/token/renew-self" {
+  capabilities = [ "update" ]
+}
+path "auth/token/lookup-self" {
+  capabilities = [ "read" ]
+}
+
+path "%[1]s/*" {
+  capabilities = [ "create", "read", "update", "delete", "list" ]
+}
+
+path "%[2]s/*" {
+  capabilities = [ "create", "read", "update", "delete", "list" ]
+}
+`, a.RootPath, a.IntermediatePath)
+
+		if a.WithSudo {
+			rules += fmt.Sprintf(`
+
+# TODO(rb): not present in docs; needed for provider swap FROM vault
```

Comment:
The docs were just updated: https://developer.hashicorp.com/consul/docs/connect/ca/vault#additional-vault-acl-policies-for-sensitive-operations